extra/src/main/java/aima/extra/environment/model/map/object/MapObject.java

```diff
@@ -0,0 +1,82 @@
+package aima.extra.environment.model.map.object;
+
+/**
+ * Created by @AdrianBZG on 2/04/17.
+ */
+
+import aima.core.util.datastructure.Tuple;
+import aima.extra.environment.view.object.ObjectView;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public abstract class MapObject implements Cloneable {
+
+    /**
+     * Internal use to have a list of all object available
+     */
+    private static ArrayList<MapObject> mapObjects = new ArrayList<>();
+
+    private Tuple<Integer, Integer> objectPosition = new Tuple<>(0, 0);
+
+
+    static {
+        // No objects at the moment so the Array is empty
+        mapObjects = new ArrayList<>(Arrays.asList());
```

Comment:
wtf? `mapObjects` gets an empty list in line 18, why here again? and why `Arrays.asList()` that is redundant too.